gpytorch/likelihoods/bernoulli_likelihood.py

```diff
@@ -20,31 +19,27 @@ class BernoulliLikelihood(Likelihood):
             p(Y=y|f)=\Phi(yf)
         \end{equation*}
     """
-    def __init__(self):
-        super().__init__()
-        self.quadrature = GaussHermiteQuadrature1D()
+    def forward(self, function_samples, **kwargs):
+        output_probs = base_distributions.Normal(0, 1).cdf(function_samples)
+        return base_distributions.Bernoulli(probs=output_probs)
 
-    def forward(self, input):
-        if not isinstance(input, MultivariateNormal):
-            raise RuntimeError(
-                "BernoulliLikelihood expects a multi-variate normally distributed latent function to make predictions"
-            )
-
-        mean = input.mean
-        var = input.variance
+    def marginal(self, function_dist, **kwargs):
+        mean = function_dist.mean
+        var = function_dist.variance
         link = mean.div(torch.sqrt(1 + var))
-        output_probs = normal_cdf(link)
-        return Bernoulli(probs=output_probs)
-
-    def variational_log_probability(self, latent_func, target):
-        likelihood_func = lambda locs: log_normal_cdf(locs.mul(target.unsqueeze(-1)))
-        res = self.quadrature(likelihood_func, latent_func)
-        return res.sum()
-
-    def pyro_sample_y(self, variational_dist_f, y_obs, sample_shape, name_prefix=""):
-        import pyro
-
-        f_samples = variational_dist_f(sample_shape)
-        y_prob_samples = torch.distributions.Normal(0, 1).cdf(f_samples)
-        y_dist = pyro.distributions.Bernoulli(y_prob_samples)
-        pyro.sample(name_prefix + "._training_labels", y_dist.independent(1), obs=y_obs)
+        output_probs = base_distributions.Normal(0, 1).cdf(link)
+        return base_distributions.Bernoulli(probs=output_probs)
+
+    def expected_log_prob(self, observations, function_dist, *params, **kwargs):
```

Comment:
Can we add a `variational_log_prob` method to the base `Likelihood` that calls `expected_log_prob` with a deprecation warning or is this hopeless? We have some custom variational stuff that this is super annoyingly breaking for.

Reply:
I can do that

Reply:
It's updated, but will the re-definition of `forward` break your custom variational stuff?

Reply:
No we don't actually use forward for this likelihood, we're never interested in p(y)